odoo/addons/base/models/ir_autovacuum.py

```diff
@@ -14,33 +16,33 @@ class AutoVacuum(models.AbstractModel):
     _name = 'ir.autovacuum'
     _description = 'Automatic Vacuum'
 
-    @api.model
-    def _gc_transient_models(self):
-        for mname in self.env:
-            model = self.env[mname]
-            if model.is_transient():
-                try:
-                    with self._cr.savepoint():
-                        model._transient_vacuum(force=True)
-                except Exception as e:
-                    _logger.warning("Failed to clean transient model %s\n%s", model, str(e))
+    def _run_vacuum_cleaner(self):
+        if not self.env.is_admin():
+            raise AccessDenied()
 
-    @api.model
-    def _gc_user_logs(self):
-        self._cr.execute("""
-            DELETE FROM res_users_log log1 WHERE EXISTS (
-                SELECT 1 FROM res_users_log log2
-                WHERE log1.create_uid = log2.create_uid
-                AND log1.create_date < log2.create_date
-            )
-        """)
-        _logger.info("GC'd %d user log entries", self._cr.rowcount)
+        for model in self.env.values():
+            cls = type(model)
+            for attr in dir(cls):
+                func = getattr(cls, attr)
+                if callable(func) and getattr(func, '_autovacuum', False):
+                    _logger.debug('Calling %s.%s', model, attr)
+                    try:
+                        getattr(model, attr)()
```

Comment:
```suggestion
                        func(model)
```

Reply:
Are we sure about it ? That's the class's function, not the model one

Reply:
Wait, I will push a commit that improves this file...